### PR TITLE
feat(Core/Misc): remove and replace ACE_Singleton

### PR DESCRIPTION
**Pull request by Winfidonarleyan**
##### CHANGES PROPOSED:
- Remove and replace ACE_Singleton in all project
- Part for dynamic linking

##### TESTS PERFORMED:
- Build and run in Windows at VS 16.3.8
- Tested in game
image

##### HOW TO TEST THE CHANGES:
- Build and run
- Enter the game and wait ~ 5 min

##### KNOWN ISSUES AND TODO LIST:
- [x] Build and run in Linux

##### Target branch(es):
- [x] Master

## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by Yehonal** [comment]
This is related to https://github.com/azerothcore/azerothcore-wotlk/issues/99 and should almost complete the rewrite of threading system.

My concerns are about the "static" instead of "std::mutex", have you tried it in a live server?

**Comment by Winfidonarleyan** [comment]
@Yehonal, yes

File: src/server/game/Maps/TransportMgr.h [hunk 1]
```diff
@@ -89,65 +88,66 @@ typedef std::map<uint32, TransportAnimation> TransportAnimationContainer;
...
-            if (itr != _transportTemplates.end())
-                return &itr->second;
-            return NULL;
-        }
+    TransportTemplate const* GetTransportTemplate(uint32 entry) const
+    {
+        TransportTemplates::const_iterator itr = _transportTemplates.find(entry);
+        if (itr != _transportTemplates.end())
+            return &itr->second;
+        return NULL;
```
Comment (Helias) [inline src/server/game/Maps/TransportMgr.h]: `nullptr` could be better here?

File: src/server/game/Maps/TransportMgr.h [hunk 2]
```diff
@@ -89,65 +88,66 @@ typedef std::map<uint32, TransportAnimation> TransportAnimationContainer;
...
-                return &itr->second;
+    TransportAnimation const* GetTransportAnimInfo(uint32 entry) const
+    {
+        TransportAnimationContainer::const_iterator itr = _transportAnimations.find(entry);
+        if (itr != _transportAnimations.end())
+            return &itr->second;
 
-            return NULL;
-        }
+        return NULL;
```
Comment (Helias) [inline src/server/game/Maps/TransportMgr.h]: `nullptr` could be better here? (too)

**Comment by Mitradis** [comment]
a few days ago i have problem due to "valgrind not found"

File: src/server/game/Maps/TransportMgr.h (hunk 1 above)
Comment (Winfidonarleyan) [reply src/server/game/Maps/TransportMgr.h]: I just replace ace_singleton

**Comment by Winfidonarleyan** [comment]
Ready to test

**Comment by Helias** [comment]
Travis build fail, try to fix it

**Comment by Bogir** [comment]
изображение

**Comment by FrancescoBorzi** [comment]
thank you @Bogir for testing this, ready to be merged

**Comment by saltyzero** [comment]
I must to say you can avoid dynamic link error with ACE_Singleton.

**Comment by Winfidonarleyan** [comment]
Yes

**Pull request merged**